### ISSUE TITLE
[EPUB 3.3] [suggestion] Note on Fixed Layout and accessibility

### ISSUE THREAD

**Issue by gregoriopellegrino** [w3c/epub-specs#1733]
Perhaps add a note in the section on Fixed Layout EPUBs indicating that there are currently no ways to produce fully accessible (i.e. EPUB Accessibility compliant) FXL EPUBs?

https://www.w3.org/TR/epub-33/#sec-fixed-layouts

**Comment by mattgarrish** [comment]
I'm not sure we'd want to do this. It's difficult and not possible for a lot of fixed layouts, but that's not really a statement for a specification to make.

Technologies and techniques are always changing, too, and it would take another revision to undo such a statement if it definitively becomes false.

**Comment by iherman** [comment]
At some point, if we indeed get an FXL A11y note published, we can add a reference to it.

**Comment by gregoriopellegrino** [comment]
My "concern" is about the [FAST Checklist](https://w3c.github.io/apa/fast/checklist.html) points:

> # If technology allows visual rendering of content
> ## Content can be resized.
> Many users need content to be displayed larger than the default, not just because of un-sharp vision but also to mitigate other visual perception difficulties such as difficulty separating foreground from background. Depending on device and situation, content can also be displayed smaller than the author expects, so needs to be resizeable even if the intended default size is suitable. Technologies should provide features to allow user resize, without experiencing problems such as pixelation, clipping, excessive scrolling, etc. Support for resizing therefore requires a variety of features to be enabled by the technology.
> ## Text presentation attributes can be changed.
> Some users with visual impairments and learning disabilities find that customizing text presentation improves their ability to distinguish letters, track lines, etc. Technologies should provide features allowing users to customize typeface, font weight, font style, line / word / letter spacing, margins, line length, justification.
> ## Changing content presentation does not render it unreadable.
> Many accessibility requirements come down to allowing users to customize presentation. When presentation is changed in a way the author or designer did not anticipate, unexpected side effects often appear that create new problems. A frequent situation is when content is resized but the region for the content is not, causing the content to be clipped. Another is when regions resize but do not reposition, making it difficult to use the content at the new scale. Change of font attributes sometimes leads to a similar problem, such as when users change to a heavier font but the space allocated for characters does not increase. Technologies should provide features to ensure that change of display attributes does not create unintended side effects.

I think that with FXL it is challenging to meet these requirements.

**Comment by mattgarrish** [comment]
As @iherman has already said, I'm more in favour of waiting for the accessibility document to be completed and then we can add a note about the accessibility challenges with a link to that document.

I'd rather be encouraging in terms of getting people to be as accessible as they can than discourage them that it's a lost cause and have them make no effort.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2022-02-17-epub#section3) on 2022-02-17

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 3. FXL Limitations.

_See github issue [epub-specs#1733](https://github.com/w3c/epub-specs/issues/1733)._




_See github issue [epub-specs#1736](https://github.com/w3c/epub-specs/issues/1736)._




**Wendy Reid:** not a lot of discussion on these, but they are still open, so we want to mention.  
… gpellegrino did review of a11y documentation last year.  
… he raised 2 issues specifically about FXL a11y.  
… one is about there being limitations on FXL a11y right now, not able to make fully accessible FXL document right now.  
… other one is about accessibility and orientation, there will be a note in best practices that blocking orientation can impede readability.  
… if no comments, both gpellgrino's comments will be reflected in note.  

**Dave Cramer:** my understanding is that the problem which cannot be avoided in making FXL accessible content is that font size is fixed.  
… is there accessibility metadata around there?.  

**Wendy Reid:** i think so, but I can't recall specific accessibility hazard text at the moment.  
… but yes, inability to modify text size, font face.  
… we will be mapping some of the statements in the best practice note to WCAG - one of these is about scrolling, i.e. having to scroll both horizontally and vertically to read content if zoomed in.  

**Dave Cramer:** i could definitely make an FXL that doesn't specify fonts, but not sure if RS would allow user to choose a font.  

**Brady Duga:** no, the whole point of FXL is that you're not supposed to change that.  

**Dave Cramer:** does the CSS positioning spec say that you can use these things to make inaccessible documents?.  

**Wendy Reid:** there is an accessibility feature call display transformability, but not corresponding accessibility hazard for no display transformability.  
… but I will ask authors of a11y spec.  

**Dave Cramer:** historically epub has allowed a lot of different ways to achieve a particular goal, and we've never taken the position that every epub must be accessible to every user.  
… for example, standard alone audio books are very useful for accessibility, but also useless to a specific segment of people.  
… this seems like we're taking a step towards saying FXL is an invalid way of presenting content to users.  

**Wendy Reid:** this is all non-normative, but i'm phrasing the note as "the limits of FXL a11y".  
… it is possible to do a lot of a11y work for a lot of people, but there are limitations that make it so that some segments will not be served.  

**Shinya Takami (高見真也):** right now we do not have requirement to make epub accessible, so it will be hard to realize accessible epubs i think.  

**Wendy Reid:** isn't a version of epub a11y 1.1 adopted as JIS standard?.  

**Murata Makoto:** it will be later this year, in aug/sep.  

**Dave Cramer:** okay, what wendyreid has proposed sounds good to me.  

</details>
